### PR TITLE
fix: handle parsing variable expression in dockerfile

### PR DESCRIPTION
**Pull request by ShauryaAg**
fixes: #336 

- Add support for parsing variable expressions in `Dockerfile` as mentioned in Dockerfile references: https://docs.docker.com/engine/reference/builder/#environment-replacement

### CODE REVIEW

**Comment by ShauryaAg** [comment]
@microsoft-github-policy-service agree

**Comment by ShauryaAg** [comment]
@microsoft-github-policy-service agree

File: src/spec-node/dockerfileUtils.ts
Comment (chrmarti) [inline src/spec-node/dockerfileUtils.ts:177]: Would it make sense to handle all 3 cases (regular, negative and positive) below? I think this could be done in a single regex?

File: src/spec-node/dockerfileUtils.ts
Comment (chrmarti) [inline src/spec-node/dockerfileUtils.ts:144]: Please add a few tests in dockerFileUtils.test.ts.

**Review by chrmarti — CHANGES_REQUESTED** [review]
Thanks for the PR! Left a few comments.

File: src/spec-node/dockerfileUtils.ts
Comment (ShauryaAg) [reply src/spec-node/dockerfileUtils.ts:150]: Fixed it! Modified it to use only one regex for all 3 cases.

File: src/spec-node/dockerfileUtils.ts
Comment (ShauryaAg) [reply src/spec-node/dockerfileUtils.ts]: Added tests.

**Comment by ShauryaAg** [comment]
Known issues:
The current implementation does not support _quoted_ expression variables, that is `${arg:+"word"}` will not be evaluated correctly in `FROM` statements as the `fromStatement` regex does not parse such `FROM` lines correctly. 

cc: @chrmarti

**Review by chrmarti — APPROVED** [review]
LGTM, thanks! (Getting a second reviewer now before we can merge.)

**Review by joshspicer — APPROVED** [review]

**Pull request merged**
